mmdet/models/detectors/d2_wrapper.py

```diff
@@ -0,0 +1,306 @@
+        name = f'{father_name}.{key.upper()}'
+        if isinstance(value, ConfigDict) or isinstance(value, dict):
+            config_list, fater_name = \
+                _to_cfgnode_list(value, config_list, name)
+        else:
+            config_list.append(name)
+            config_list.append(value)
+
+    return config_list, father_name
+
+
+def add_d2_pred_to_datasample(data_samples: SampleList,
+                              d2_results_list: list) -> SampleList:
+    """Convert the Detectron2's result to DetDataSample.
+
+    Args:
+        data_samples (list[:obj:`DetDataSample`]): The batch
+            data samples. It usually includes information such
+            as `gt_instance` or `gt_panoptic_seg` or `gt_sem_seg`.
+        d2_results_list (list): The list of the results of Detectron2's model.
+
+    Returns:
+        list[:obj:`DetDataSample`]: Detection results of the
+        input images. Each DetDataSample usually contain
+        'pred_instances'. And the ``pred_instances`` usually
+        contains following keys.
+
+        - scores (Tensor): Classification scores, has a shape
+          (num_instance, )
+        - labels (Tensor): Labels of bboxes, has a shape
+          (num_instances, ).
+        - bboxes (Tensor): Has a shape (num_instances, 4),
+          the last dimension 4 arrange as (x1, y1, x2, y2).
+    """
+    assert len(data_samples) == len(d2_results_list)
+    for data_sample, d2_results in zip(data_samples, d2_results_list):
+        d2_instance = d2_results['instances']
+
+        results = InstanceData()
+        results.bboxes = d2_instance.pred_boxes.tensor
+        results.scores = d2_instance.scores
+        results.labels = d2_instance.pred_classes
+
+        if d2_instance.has('pred_masks'):
+            results.masks = d2_instance.pred_masks
+        data_sample.pred_instances = results
+
+    return data_samples
+
+
+@MODELS.register_module()
+class Detectron2Wrapper(BaseDetector):
+    """Wrapper of a Detectron2 model. Input/output formats of this class follow
+    MMDetection's convention, so a Detectron2 model can be trained and
+    evaluated in MMDetection.
+
+    Args:
+        d2_detector (:obj:`ConfigDict` or dict): The module config of
+            Detectron2.
+        bgr_to_rgb (bool): whether to convert image from BGR to RGB.
+            Defaults to False.
+        rgb_to_bgr (bool): whether to convert image from RGB to RGB.
+            Defaults to False.
+        data_preprocessor (:obj:`ConfigDict` or dict, optional): Config of
+            :class:`DetDataPreprocessor` to process the input data.
+            Defaults to None.
+        init_cfg (:obj:`ConfigDict` or dict, optional): the config to control
+            the initialization. Defaults to None.
+    """
+
+    def __init__(self,
+                 d2_detector: ConfigType,
+                 bgr_to_rgb: bool = False,
+                 rgb_to_bgr: bool = False,
+                 data_preprocessor: OptConfigType = None,
```

Comment:
data_preprocessor does not seem to require user settings, can it be deleted directly?